### PR TITLE
Provide support for ACS5 Subject Tables 

### PR DESCRIPTION
**Pull request by Detinator10**
The code for this pull request comes from @grahamalama . I simply made a small edit to the dataset variable to change it to acs5/subject as it and pulled the latest version of census before making these changes. I've tested the subject client personally on my own project and it appears to work just fine. The reason I am submitting this pull request is because #60 is held up with merge conflicts in the README and is out of date/has one error related to the dataset variable. All credit should go to @grahamalama though

### CODE REVIEW

**Comment by grahamalama** [comment]
Glad to see this getting some attention!

I looked into this a bit and figured out why your dataset property worked and mine didn't. 

When I submitted my PR, I needed data from 2015. If you try to `get` data where `year=2015` with your change, I think it will fail, but will succeed on my branch. But then from 2016 on, it will work with the `dataset` property you suggested. I made [this gist](https://gist.github.com/grahamalama/8af8a3dc8d8d3dbdf4f60f0c02533618) to play around with this.

This looks like it's a symptom of the `_switch_endpoints` method of `ACSClient`. We always want `acs` to appear in the URL for querying the ACS5 subject tables, so we shouldn't actually switch endpoints.

I like your dataset property better, so I'd suggest modifying the `ACS5SubjectClient` class so that it keeps the same endpoint for each year. I kind of hate it, but a quick and dirty way would be to just override `_switch_endpoints` with:

```python
def _switch_endpoints(self, year):
    self.endpoint_url = 'https://api.census.gov/data/%s/acs/%s'
    self.definitions_url = 'https://api.census.gov/data/%s/acs/%s/variables.json'
    self.definition_url = 'https://api.census.gov/data/%s/acs/%s/variables/%s.json'
```

**Comment by Detinator10** [comment]
> Glad to see this getting some attention!
> 
> I looked into this a bit and figured out why your dataset property worked and mine didn't.
> 
> When I submitted my PR, I needed data from 2015. If you try to `get` data where `year=2015` with your change, I think it will fail, but will succeed on my branch. But then from 2016 on, it will work with the `dataset` property you suggested. I made [this gist](https://gist.github.com/grahamalama/8af8a3dc8d8d3dbdf4f60f0c02533618) to play around with this.
> 
> This looks like it's a symptom of the `_switch_endpoints` method of `ACSClient`. We always want `acs` to appear in the URL for querying the ACS5 subject tables, so we shouldn't actually switch endpoints.
> 
> I like your dataset property better, so I'd suggest modifying the `ACS5SubjectClient` class so that it keeps the same endpoint for each year. I kind of hate it, but a quick and dirty way would be to just override `_switch_endpoints` with:
> 
> ```python
> def _switch_endpoints(self, year):
>     self.endpoint_url = 'https://api.census.gov/data/%s/acs/%s'
>     self.definitions_url = 'https://api.census.gov/data/%s/acs/%s/variables.json'
>     self.definition_url = 'https://api.census.gov/data/%s/acs/%s/variables/%s.json'
> ```

I think I added the code you were talking about with switch endpoints in my latest commit. I'm going to run some tests now to make sure

**Comment by Detinator10** [comment]
Ok, I tested the new code with both no year specified and 2015 and both worked. Thank you @grahamalama for catching that!

File: census/core.py [hunk 1]
```diff
@@ -518,6 +527,7 @@ def __init__(self, key, year=None, session=None):
         self.acs5 = ACS5Client(key, year, session)
         self.acs3 = ACS3Client(key, year, session)
         self.acs1 = ACS1Client(key, year, session)
+        self.acs5subject = ACS5SubjectClient(key, year, session)
```
Comment (jeancochrane) [inline census/core.py:530]: The naming convention in this library tends to prefer acronyms, e.g. `acs5dp` for data profiles and `acs5sf` for summary files. To match this convention, how about the name `acs5st` for this client?

**Comment by Detinator10** [comment]
I made the edits you discussed Jean, thank you. I'll check back in a while to make sure the CI build finishes and is successful.

**Comment by Detinator10** [comment]
Well, as I posted that, the CI finished, looks good to me. Let me know if there's anything else I should do

File: census/tests/test_census.py [hunk 2]
```diff
@@ -171,7 +176,7 @@ def test_acs5(self):
 
         self.check_endpoints('acs5', tests)
 
-    def test_acs1dp(self):
+    def test_acs5st(self):
```
Comment (jeancochrane) [inline census/tests/test_census.py:179]: One tiny mixup here:

```suggestion
    def test_acs1dp(self):
```

File: census/tests/test_census.py [hunk 3]
```diff
@@ -181,6 +186,18 @@ def test_acs1dp(self):
         )
 
         self.check_endpoints('acs1dp', tests)
+    
+    def test_acs1dp(self):
```
Comment (jeancochrane) [inline census/tests/test_census.py:190]: Also here:

```suggestion
    def test_acs5st(self):
```

**Comment by Detinator10** [comment]
Ok, I think I got it

**Pull request merged**